make/tests

```diff
@@ -1,3 +1,59 @@
+ifdef STANC2
+TEST_STANC=test/test-models/stanc2$(EXE)
+else
+TEST_STANC=bin/stanc$(EXE)
+endif
+
+##
+# Build/download stanc3
+##
+
+STANC_DL_RETRY = 5
+STANC_DL_DELAY = 10
+
+ifeq ($(OS),Windows_NT)
+ OS_TAG := windows
+else ifeq ($(OS),Darwin)
+ OS_TAG := mac
+else ifeq ($(OS),Linux)
+ OS_TAG := linux
+endif
+
+ifneq ($(STANC3),)
+  
+  # build stanc3 from local installation
+  bin/stanc$(EXE) : $(call findfiles,$(STANC3)/src/,*.ml*) $(STANC#)
+	@mkdir -p $(dir $@)
+	cd $(STANC3) && echo "--- Rebuilding stanc ---\n" && dune build @install
+	cp $(STANC3)/_build/default/src/stanc/stanc.exe $@
+	
+else ifneq ($(STANC3_TEST_BIN_URL),)
+
+# download stanc3 build from specific branch
+ifeq ($(OS_TAG),windows)
+    bin/stanc$(EXE) :
+	@mkdir -p $(dir $@)
+	$(shell echo "curl -L $(STANC3_TEST_BIN_URL)/bin/$(OS_TAG)-stanc -o bin/stanc$(EXE) --retry $(STANC_DL_RETRY) --retry-delay $(STANC_DL_DELAY)")
+else
+    bin/stanc$(EXE) :
+	@mkdir -p $(dir $@)
+	curl -L $(STANC3_TEST_BIN_URL)/bin/$(OS_TAG)-stanc -o bin/stanc$(EXE) --retry $(STANC_DL_RETRY) --retry-delay $(STANC_DL_DELAY)
+	chmod +x bin/stanc$(EXE)
+endif # end download if/else
+
+else ifeq ($(OS_TAG),windows)
+# get latest stanc3 - Windows
+    bin/stanc$(EXE) :
+	@mkdir -p $(dir $@)
+	$(shell echo "curl -L https://github.com/stan-dev/stanc3/releases/download/nightly/$(OS_TAG)-stanc -o bin/stanc$(EXE) --retry $(STANC_DL_RETRY) --retry-delay $(STANC_DL_DELAY)")
+else
+# get latest stanc3 - Linux & MacOS
+    bin/stanc$(EXE) :
+	@mkdir -p $(dir $@)
+	curl -L https://github.com/stan-dev/stanc3/releases/download/nightly/$(OS_TAG)-stanc -o bin/stanc$(EXE) --retry $(STANC_DL_RETRY) --retry-delay $(STANC_DL_DELAY)
+	chmod +x bin/stanc$(EXE)
+endif
+
```

Comment:
All this logic was copied from cmdstan, removing the option of using the "release binary" (cmdstan release tarball comes with bin/windows-stanc, bin/linux-stanc etc). Stan release tarball does not come with it, so no sense of having that logic here.